cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureNetworkConnector.java

```diff
@@ -231,17 +230,20 @@ public void createProviderSpecificNetworkResources(NetworkResourcesCreationReque
             networkView.setExistingNetwork(request.isExistingNetwork());
             networkView.setNetworkId(request.getNetworkId());
             networkView.setResourceGroupName(request.getNetworkResourceGroup());
-            azureDnsZoneService.checkOrCreateDnsZones(authenticatedContext, azureClient, networkView, resourceGroup, tags);
-            azureNetworkLinkService.checkOrCreateNetworkLinks(authenticatedContext, azureClient, networkView, resourceGroup, tags);
+            Set<AzurePrivateDnsZoneServiceEnum> servicesWithExistingDnsZone = getServicesWithExistingDnsZone(request);
+            azureDnsZoneService.checkOrCreateDnsZones(authenticatedContext, azureClient, networkView, resourceGroup, tags, servicesWithExistingDnsZone);
+            azureNetworkLinkService.checkOrCreateNetworkLinks(authenticatedContext, azureClient, networkView, resourceGroup, tags, servicesWithExistingDnsZone);
         } else {
-            if (request.isExistingPrivateDnsZone()) {
-                LOGGER.debug("The private DNS zone '{}' already exists, nothing to do. ", request.getExistingPrivateDnsZoneId());
-            } else {
-                LOGGER.debug("Private endpoints are disabled, nothing to do.");
-            }
+            LOGGER.debug("Private endpoints are disabled, nothing to do.");
```

Comment:
pls cover this with unit test

Reply:
I added a test to cover that branch as well.